web/client/plugins/TOC/components/StyleBasedWMSJsonLegend.jsx

```diff
@@ -0,0 +1,153 @@
+        }
+    }
+
+    getLegendData() {
+        let jsonLegendUrl = this.getUrl(this.props);
+        if (!jsonLegendUrl) {
+            this.setState({ error: true });
+            return;
+        }
+        this.setState({ loading: true });
+        getJsonWMSLegend(jsonLegendUrl).then(data => {
+            this.setState({ jsonLegend: data[0], loading: false });
+        }).catch(() => {
+            this.setState({ error: true, loading: false });
+        });
+    }
+
+    getScale = (props) => {
+        if (props.scales && props.currentZoomLvl !== undefined && props.scaleDependent) {
+            const zoom = Math.round(props.currentZoomLvl);
+            const scale = props.scales[zoom] ?? props.scales[props.scales.length - 1];
+            return Math.round(scale);
+        }
+        return null;
+    };
+    getUrl = (props, urlIdx) => {
+        if (props.layer && props.layer.type === "wms" && props.layer.url) {
+            const layer = props.layer;
+            const idx = !isNil(urlIdx) ? urlIdx : isArray(layer.url) && Math.floor(Math.random() * layer.url.length);
+
+            const url = isArray(layer.url) ?
+                layer.url[idx] :
+                layer.url.replace(/[?].*$/g, '');
+
+            let urlObj = urlUtil.parse(url);
+
+            const cleanParams = clearNilValuesForParams(layer.params);
+            const scale = this.getScale(props);
+            let query = assign({}, {
+                service: "WMS",
+                request: "GetLegendGraphic",
+                format: "application/json",
+                height: props.legendHeight,
+                width: props.legendWidth,
+                layer: layer.name,
+                style: layer.style || null,
+                version: layer.version || "1.3.0",
+                SLD_VERSION: "1.1.0",
+                LEGEND_OPTIONS: props.legendOptions
+            }, layer.legendParams || {},
+            props.language && layer.localizedLayerStyles ? {LANGUAGE: props.language} : {},
+            addAuthenticationToSLD(cleanParams || {}, props.layer),
+            cleanParams && cleanParams.SLD_BODY ? {SLD_BODY: cleanParams.SLD_BODY} : {},
+            scale !== null ? { SCALE: scale } : {});
+            addAuthenticationParameter(url, query);
+
+            return urlUtil.format({
+                host: urlObj.host,
+                protocol: urlObj.protocol,
+                pathname: urlObj.pathname,
+                query: query
+            });
+        }
+        return '';
+    }
+    renderRules = (rules) => {
+        const isLegendFilterIncluded = this.props.layer?.layerFilter?.filters?.find(f=>f.id === 'interactiveLegend');
+        const prevFilter = isLegendFilterIncluded ? isLegendFilterIncluded?.filters?.[0]?.id : '';
+        const filterWMSLayerHandler = (filter) => {
+            if (!filter) return;
+            this.props.onLayerFilterByLegend(this.props.layer.id, 'layers', filter === prevFilter ? '' : filter);
```

Comment:
Here we should remove onLayerFilterByLegend and use the passed onChange, I would expect something like this:

```
const newLayerFilter = updateLayerLegendFilter(layer?.layerFilter, filter);
this.props.onChange({ layerFilter: newLayerFilter });

```

Inside the `updateLayerLegendFilter` utils we will place all the logic to update the layerFilter. Note queryform expect an empty filter with specific property available to make it work propertly, from my investigation in review this default properties should be available:

```js
const layerFilter = {
    groupFields: [
        {
            id: 1,
            logic: 'OR',
            index: 0
        }
    ],
    filterFields: [],
    attributePanelExpanded: true,
    spatialPanelExpanded: true,
    crossLayerExpanded: true,
    crossLayerFilter: {
        attribute: 'the_geom'
    },
    spatialField: {
        method: null,
        operation: 'INTERSECTS',
        geometry: null,
        attribute: 'the_geom'
    },
```

- groupFields is needed to show the first row in the attribute filter
- most of the other are needed to make the save button enabled after apply a filter